src/nrepl/index.ts

```diff
@@ -393,21 +431,32 @@ export class NReplSession {
     const id = this.client.nextId;
     return new Promise<void>((resolve, reject) => {
       this.messageHandlers[id] = resultHandler(resolve, reject);
-      this.client.write({
+      const msg = {
         op: 'interrupt',
         session: this.sessionId,
         'interrupt-id': interruptId,
         id,
-      });
+      };
+      if (this.supports('interrupt')) {
+        this.client.write(msg);
+      } else {
+        log(msg, Direction.ClientToServerNotSupported);
+        resolve(undefined);
+      }
     });
   }
 
   stdin(message: string) {
-    this.client.write({
+    const msg = {
       op: 'stdin',
       stdin: message,
       session: this.sessionId,
-    });
+    };
+    if (this.supports('stdin')) {
```

Comment:
I think it would be better to check for support inside of the `write` function, rather than having to check in every function that calls `write`. Then this if check + log statement (if the op isn't supported), is only in one place.

Reply:
Maybe make `write` return `true` if the op is supported or `undefined` if it's not, then return the result of `write` in these op calling functions.

Reply:
I had it in the write function at first. But then it looked wrong to do a bunch of maybe-writes, and I also had some problems with it. The functions need to resolve with the result of the op, so I still had logic in the functions checking the result of the write...

But I reverted to this way without investigating the options to press on with the centralized approach. I guess if we pass on the resultResolver to write, it can handle this and we would have even less duplicated code. I'll have a look, but maybe we need to write it up on the todo list.

Reply:
Yeah, I'm pretty sure it can be done, but all my attempts get very messy so far. Someone with better command over JavaScript than I have might see what should be done.